File: src/test/java/cz/solutions/cockroach/TabularExchangeRateProviderTest.java
package cz.solutions.cockroach;

import org.joda.time.LocalDate;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

class TabularExchangeRateProviderTest {

    @Test
    void canParseHardcoded() throws Exception {
        TabularExchangeRateProvider rateProvider = TabularExchangeRateProvider.hardcoded();

        assertThat(rateProvider.rateAt(LocalDate.parse("2021-05-16")), is(21.024));
        assertThat(rateProvider.rateAt(LocalDate.parse("2021-05-14")), is(21.024));
        assertThat(rateProvider.rateAt(LocalDate.parse("2022-05-15")), is(23.825));
        assertThat(rateProvider.rateAt(LocalDate.parse("2023-05-15")), is(21.671));
        assertThat(rateProvider.rateAt(LocalDate.parse("2024-05-15")), is(22.861));
    }
}

File: src/test/kotlin/cz/solutions/cockroach/TabularExchangeRateProviderTest.kt
package cz.solutions.cockroach

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.joda.time.LocalDate
import org.junit.jupiter.api.Test

internal class TabularExchangeRateProviderTest {

    @Test
    fun `can parse hardcoded`() {
        val rateProvider = TabularExchangeRateProvider.hardcoded()
        assertThat(rateProvider.rateAt(LocalDate.parse("2021-05-16")), `is`(21.024))
        assertThat(rateProvider.rateAt(LocalDate.parse("2021-05-14")), `is`(21.024))
        assertThat(rateProvider.rateAt(LocalDate.parse("2022-05-15")), `is`(23.825))
        assertThat(rateProvider.rateAt(LocalDate.parse("2023-05-15")), `is`(21.671))
        assertThat(rateProvider.rateAt(LocalDate.parse("2024-05-15")), `is`(22.861))
    }
}